torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -414,9 +414,15 @@ def _pre_forward(
     # the `grad_fn` is mutated.
     _register_post_backward_hooks(state, handles)
 
-    # Recursively convert args and kwargs to specified precision.
-    input_dtype: Optional[torch.dtype] = state.mixed_precision.param_dtype
-    if state.mixed_precision.cast_forward_inputs:
+    should_cast_forward = True
+    for handle in state._handles:
+        if handle._force_full_precision:
+            should_cast_forward = False
+            break
+
+    if should_cast_forward and state.mixed_precision.cast_forward_inputs:
```

Comment:
Is this branch covered by unit tests?

Reply:
Yes, if this wasn't added, there would be a dtype mismatch because inputs are in lower precision and parameters are not.

Reply:
Just in case: I meant I did not see passing `cast_forward_inputs=True` in the added unit tests.

Reply:
good point, this is false by default, will add to tests